Sources/SKTestSupport/IndexedSingleSwiftFileWorkspace.swift

```diff
@@ -0,0 +1,125 @@
+
+    let testFileURL = testWorkspaceDirectory.appendingPathComponent("test.swift")
+    let indexURL = testWorkspaceDirectory.appendingPathComponent("index")
+    let indexDBURL = testWorkspaceDirectory.appendingPathComponent("index-db")
+    let toolchain = ToolchainRegistry.shared.default!
+    guard let swiftc = toolchain.swiftc?.asURL else {
+      throw Error.swiftcNotFound
+    }
+
+    // Create workspace with source file and compile_commands.json
+
+    try extractMarkers(markedText).textWithoutMarkers.write(to: testFileURL, atomically: false, encoding: .utf8)
+
+    let compilerArguments: [String] = [
+      testFileURL.path,
+      "-sdk", TibsBuilder.defaultSDKPath!,
+      "-index-store-path", indexURL.path,
+      "-index-ignore-system-modules",
+      "-o", testWorkspaceDirectory.appendingPathComponent("test.out").path
+    ]
+
+    let compilationDatabase = JSONCompilationDatabase(
+      [
+        JSONCompilationDatabase.Command(
+          directory: testWorkspaceDirectory.path,
+          filename: testFileURL.path,
+          commandLine: [swiftc.path] + compilerArguments
+        )
+      ]
+    )
+    let encoder = JSONEncoder()
+    encoder.outputFormatting = .prettyPrinted
+    try encoder.encode(compilationDatabase).write(to: testWorkspaceDirectory.appendingPathComponent("compile_commands.json"))
+
+    // Run swiftc to build the index store
+    try ProcessRunner(executableURL: swiftc, arguments: compilerArguments)
+      .run(verbose: false)
+
+    // Create the test client
+    var options = SourceKitServer.Options.testDefault
+    options.indexOptions = IndexOptions(
+      indexStorePath: try AbsolutePath(validating: indexURL.path),
+      indexDatabasePath: try AbsolutePath(validating: indexDBURL.path)
+    )
+    self.testClient = try await TestSourceKitLSPClient(
+      serverOptions: options,
+      workspaceFolders: [
+        WorkspaceFolder(uri: DocumentURI(testWorkspaceDirectory))
+      ],
+      cleanUp: {
+        try? FileManager.default.removeItem(at: testWorkspaceDirectory)
+      }
+    )
+
+    // Wait for the indexstore-db to finish indexing
+    _ = try await testClient.send(PollIndexRequest())
+
+    // Open the document
+    self.fileURI = DocumentURI(testFileURL)
+    self.positions = testClient.openDocument(markedText, uri: fileURI)
+  }
+}
+
+
+
+fileprivate extension URL {
+  /// Assuming this is a file URL, resolves all symlinks in the path.
+  ///
+  /// - Note: We need this because `URL.resolvingSymlinksInPath()` not only resolves symlinks but also standardizes the
+  ///   path by stripping away `private` prefixes. Since sourcekitd is not performing this standardization, using
+  ///   `resolvingSymlinksInPath` can lead to slightly mismatched URLs between the sourcekit-lsp response and the test
+  ///   assertion.
+  var realpath: URL {
+    return self.path.withCString { path in
+      guard let realpath = Darwin.realpath(path, nil) else {
+        return self
+      }
```

Comment:
Ah, forgot about that again. Luckily, we can just use `resolvingSymlinksInPath()` on those platforms because it doesn’t have the `private`-stripping issue